### ISSUE TITLE
Rule suggestion: newline-before-expect

### ISSUE THREAD

**Issue by cbrwizard** [tlvince/eslint-plugin-jasmine#50]
I suggest adding a new rule: `newline-before-expect`, which will serve the same purpose [`newline-before-return`](http://eslint.org/docs/rules/newline-before-return) does, except for expect.

`expect` usually is like `return` but for specs [: Clearly delineating where a spec is running an expectation can greatly increase the readability and clarity of the code.

For the use case check [this issue](https://github.com/eslint/eslint/issues/5734) on eslint.

I suggest adding it with the following options:

`everyExpect`: tells to add a new-line before every `expect` in a spec. Sometimes developers put multiple `expect`-s in one spec (btw we should add a rule to limit an `it` function so that it has only one expectation) and this option tells should a new line be added before every expect or only before the first in the series of lines with `expect`. Default: false.

`minBlockSize`: tells when to force a newline before expect. Example:

``` js
  it('returns a passed keyCode', function() {
    var keyboardEvent = createEvent({type: 'keyup', keyCode: 40});

    expect(keyboardEvent.which).toBe(40);
  });
```

In such a small function I'd rather not have a newline before `expect` keyword. However, in this function I would like to have one.

``` js
  it('returns whatever getEventCharCode returns', function() {
    getEventCharCode.mockReturnValue(9001);
    var keyboardEvent = createEvent({type: 'keypress', charCode: 50});

    expect(keyboardEvent.which).toBe(9001);
  });
```

Default: 0 (forced in every function).

If you're interested, I can add more info for clarity and even implement it afterwards. What do you think?

**Comment by tlvince** [comment]
This is not something I'd use personally, but would be happy to include if there's interest for it. I'd suggest the rule should be opt-in (disabled by default).

**Comment by alecxe** [comment]
Nice idea! I would actually enable that in my projects, but I agree that we should not enforce it.

**Comment by burabure** [comment]
hey @cbrwizard ! want to help with this? I'd love to pair if you like

**Comment by DianaSuvorova** [comment]
Hey @burabure, do you want me to do this?

**Comment by burabure** [comment]
@DianaSuvorova sure, go ahead!

**Comment by DianaSuvorova** [comment]
@burabure, this can be closed 😄
